Sources/protoc-gen-swift/FileGenerator.swift

```diff
@@ -247,7 +250,7 @@ class FileGenerator {
     }
 
     func generateOutputFile(printer p: inout CodePrinter, context: Context) {
-        let inputFilename = descriptor.name ?? "<No name>";
+        let inputFilename = descriptor.hasName ? descriptor.name : "<No name>"
```

Comment:
When I was doing the output names, I was debating stuff like this.  Protoc really never should send us a descriptor without names, so it might make sense to eventually do a sweep for stuff like this and change the code to just abort.  That's what most of the C++ generators do when they get input that doesn't make sense.